document-store/src/main/java/org/hypertrace/core/documentstore/mongo/query/parser/MongoGroupTypeExpressionParser.java

```diff
@@ -151,10 +151,12 @@ public static boolean isFunctionExpressionSelectionWithGroupBy(
         && (Boolean) selectionSpec.getExpression().accept(FUNCTION_EXPRESSION_CHECKER);
   }
 
+  @SuppressWarnings("unchecked")
   public static List<String> getGroupByAliases(final List<GroupTypeExpression> expressions) {
     return expressions.stream()
-        .map(expression -> (String) expression.accept(GROUP_BY_ALIAS_GETTER))
-        .filter(Objects::nonNull)
+        .map(expression -> (Optional<String>) expression.accept(GROUP_BY_ALIAS_GETTER))
+        .filter(Optional::isPresent)
+        .map(Optional::get)
```

Comment:
Nit: Can also be simplified to.
```suggestion
        .flatMap(Optional::stream)
```